insights/parsers/alternatives.py

```diff
@@ -0,0 +1,159 @@
+        '/usr/lib/jvm/java-1.7.0-openjdk-1.7.0.111-2.6.7.2.el7_2.x86_64/jre/bin/java'
+        >>> java.paths[0]['priority']
+        1700111
+        >>> java.paths[2]['slave']['ControlPanel']
+        '/usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/ControlPanel'
+"""
+
+    def parse_content(self, content):
+        """
+        Parse the output of the ``alternatives`` command.
+        """
+        self.program = None
+        self.status = None
+        self.link = None
+        self.best = None
+        self.paths = []
+        current_path = {}
+        # Set up instance variable
+
+        for line in content:
+            words = line.split(None)
+            if ' - status is' in line:
+                # line: 'java - status is auto'.
+                # Store old program
+                if self.program:
+                    # alternatives only displays one program, so finding
+                    # this line again is an error.
+                    raise ParseException(
+                        "Program line for {newprog} found in output for {oldprog}".format(
+                            newprog=words[0], oldprog=self.program
+                        )
+                    )
+                # Set up new program data
+                self.program = words[0]
+                self.status = words[4][:-1]  # remove trailing .
+                self.alternatives = []
+                current_path = {}
+            elif not self.program:
+                # Lines before 'status is' line are ignored
+                continue
+            elif line.startswith(' link currently points to ') and len(words) == 5:
+                # line: ' link currently points to /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.111-1.b15.el7_2.x86_64/jre/bin/java'
+                self.link = words[4]
+            elif ' - priority ' in line and len(words) == 4 and words[3].isdigit():
+                # line: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/java - priority 16091
+                # New path - save current path if set
+                if current_path:
+                    self.paths.append(current_path)
+                current_path = {
+                    'path': words[0],
+                    'priority': int(words[3]),
+                    'slave': {},
+                }
+            elif line.startswith(' slave ') and len(words) == 3 and current_path:
+                # line: ' slave ControlPanel: /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/ControlPanel'
+                current_path['slave'][words[1][:-1]] = words[2]  # remove final : from program
+            elif line.startswith("Current `best' version is ") and len(words) == 5:
+                # line: 'Current `best' version is /usr/lib/jvm/jre-1.6.0-ibm.x86_64/bin/java.'
+                self.best = words[4][:-1]  # strip trailing . from path
+
+        # Store program's current path if both are available
+        if self.program and current_path:
+                self.paths.append(current_path)
```

Comment:
I'm not sure whether you think they should be indented or outdented, but this needs to occur even if the output does not contain the current 'best' version.

However, I've realised that I can simplify this parsing a bit - let me know what you think of it now.